src/components/structures/ScrollPanel.js

```diff
@@ -131,8 +148,53 @@ module.exports = React.createClass({
         if (sn.scrollTop < sn.clientHeight) {
             // there's less than a screenful of messages left - try to get some
             // more messages.
-            this.props.onFillRequest(true);
+            this._maybeFill(true);
         }
+        if (sn.scrollTop > sn.scrollHeight - sn.clientHeight * 2) {
```

Comment:
Magical calculation is magical. What is `clientHeight`? Why `*2`? Please either in-line comment or convert this to a well-named function.